File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        int low = 0;
        int high = s.length() - 1;
        char[] charArray = s.toCharArray();

        while (low < high) {
            if (isVowel(charArray[low]) && isVowel(charArray[high])) {
                char temp = charArray[low];
                charArray[low] = charArray[high];
                charArray[high] = temp;
                low++;
                high--;
            } else if (isVowel(charArray[low])) {
                high--;
            } else {
                low++;
            }
        }

        return new String(charArray);
    }

    static boolean isVowel(char c) {
        return "aeiouAEIOU".indexOf(c) != -1;
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
       vowels = ['a','e','i','o','u','A','E','I','O','U']
       low = 0
       high = len(s)-1
       strList = list(s)
       while(low<high):
         if(s[low] in vowels and s[high] in vowels):
            strList[low] = s[high]
            strList[high] = s[low]
            low+=1
            high-=1
         elif(s[low] in vowels):
            high-=1
         else:
            low+=1
       return "".join(strList)

        

 